homeassistant/components/trest_solar/sensor.py

```diff
@@ -0,0 +1,133 @@
+"""Entity that represents the trest solar controller and its data values."""
+
+from datetime import datetime
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import TrestConfigEntry
+from .const import DOMAIN
+from .coordinator import TrestDataCoordinator
+
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="battery_discharge",
+        name="Battery Discharge",
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement="kW",
+        icon="mdi:battery-arrow-down",
+    ),
+    SensorEntityDescription(
+        key="battery_charge",
+        name="Battery Charge",
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement="kW",
+        icon="mdi:battery-arrow-up",
+    ),
+    SensorEntityDescription(
+        key="battery_capacity",
+        name="Battery Capacity",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement="%",
+        icon="mdi:battery",
+    ),
+    SensorEntityDescription(
+        key="battery_stored_power",
+        name="Battery Stored Power",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement="kW",
+        icon="mdi:power",
+    ),
+    SensorEntityDescription(
+        key="total_load_active_power",
+        name="Total Load Active Power",
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement="kW",
+        icon="mdi:flash",
+    ),
+    SensorEntityDescription(
+        key="realtime_solar",
+        name="Realtime Solar",
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement="kW",
+        icon="mdi:white-balance-sunny",
+    ),
+    SensorEntityDescription(
+        key="timestamp",
+        name="Timestamp",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        icon="mdi:clock",
+    ),
```

Comment:
We don't allow sensors that show the last updated time since HA already stores that data.

Reply:
This is kind of needed because this is the timestamp of when the data was pushed from the IoT device that fetches the data and pushes it to the cloud.